### PR TITLE
Pylint pass

### PR DESCRIPTION
**Pull request by QB3**
to solve #9

### CODE REVIEW

**Comment by QB3** [comment]
do you still have pylint error messages @mathurinm ?

**Comment by mathurinm** [comment]
```
(base) ➜  CLaR git:(pylint) pylint clar                           
************* Module clar.utils
clar/utils.py:163:0: C0325: Unnecessary parens after 'assert' keyword (superfluous-parens)
clar/utils.py:207:9: R1714: Consider merging these comparisons with "in" to "pb_name in ('CLAR', 'NNCVX')" (consider-using-in)
clar/utils.py:278:31: W0622: Redefining built-in 'ord' (redefined-builtin)
clar/utils.py:294:37: W0622: Redefining built-in 'ord' (redefined-builtin)
clar/utils.py:300:49: W0622: Redefining built-in 'ord' (redefined-builtin)
************* Module clar.duality_gap
clar/duality_gap.py:13:28: W0613: Unused argument 'sigma_min' (unused-argument)
clar/duality_gap.py:125:4: W0612: Unused variable 'n_sensors' (unused-variable)
clar/duality_gap.py:125:15: W0612: Unused variable 'n_times' (unused-variable)
************* Module clar.solvers
clar/solvers.py:351:18: C0326: Exactly one space required after comma
        X, Y, B, R,  S_inv_R, S_inv_X,
                  ^ (bad-whitespace)
clar/solvers.py:18:19: W0613: Unused argument 'B0' (unused-argument)
clar/solvers.py:21:8: W0613: Unused argument 'verbose' (unused-argument)
clar/solvers.py:59:8: W0613: Unused argument 'n_nncvx_iter' (unused-argument)
clar/solvers.py:165:7: R1714: Consider merging these comparisons with "in" to "pb_name in ('CLAR', 'mrce')" (consider-using-in)
clar/solvers.py:200:9: R1714: Consider merging these comparisons with "in" to "pb_name in ('MTL', 'MTLME')" (consider-using-in)
clar/solvers.py:205:20: E1101: Module 'numpy.linalg' has no 'pinvh' member; maybe 'pinv'? (no-member)
clar/solvers.py:239:17: R1714: Consider merging these comparisons with "in" to "pb_name in ('MTL', 'MTLME')" (consider-using-in)
clar/solvers.py:280:37: E1124: Argument 'order' passed by position and keyword in method call (redundant-keyword-arg)
clar/solvers.py:213:4: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
clar/solvers.py:297:17: R1714: Consider merging these comparisons with "in" to "pb_name in ('MTL', 'MTLME')" (consider-using-in)
clar/solvers.py:153:0: R0912: Too many branches (35/12) (too-many-branches)
clar/solvers.py:153:0: R0915: Too many statements (121/50) (too-many-statements)
clar/solvers.py:341:9: R1714: Consider merging these comparisons with "in" to "pb_name in ('MTL', 'MTLME')" (consider-using-in)
clar/solvers.py:329:22: W0613: Unused argument 'R' (unused-argument)
clar/solvers.py:357:17: R1714: Consider merging these comparisons with "in" to "pb_name not in ('MTL', 'MTLME')" (consider-using-in)
clar/solvers.py:410:17: E1101: Module 'numpy.linalg' has no 'pinvh' member; maybe 'pinv'? (no-member)
clar/solvers.py:431:29: I1101: Module 'sklearn.linear_model.cd_fast' has no 'enet_coordinate_descent_gram' member, but source is unavailable. Consider adding this module to extension-pkg-whitelist if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
clar/solvers.py:391:49: W0613: Unused argument 'mode' (unused-argument)
clar/solvers.py:391:60: W0613: Unused argument 'tol' (unused-argument)
clar/solvers.py:392:23: W0613: Unused argument 'sigmamin' (unused-argument)
clar/solvers.py:392:52: W0613: Unused argument 'verbose' (unused-argument)
clar/solvers.py:393:8: W0613: Unused argument 'return_costs' (unused-argument)
clar/solvers.py:394:8: W0613: Unused argument 'return_n_iter' (unused-argument)
************* Module clar.generate_data
clar/generate_data.py:35:23: W0613: Unused argument 'seed' (unused-argument)
clar/generate_data.py:45:40: W0613: Unused argument 'n_active' (unused-argument)
clar/generate_data.py:48:16: W0612: Unused variable 'n_sources' (unused-variable)
************* Module clar.tests.test_clar
clar/tests/test_clar.py:15:8: W0613: Unused argument 'dictionary_type' (unused-argument)
************* Module clar.tests.test_sgcl
clar/tests/test_sgcl.py:15:8: W0613: Unused argument 'dictionary_type' (unused-argument)
************* Module clar.data.artificial
clar/data/artificial.py:77:0: C0330: Wrong hanging indentation (add 4 spaces).
        X, B_star, S_star, n_epochs=n_epochs,
        ^   | (bad-continuation)
clar/data/artificial.py:78:0: C0330: Wrong hanging indentation (add 4 spaces).
        n_active=n_active, SNR=SNR, seed=seed)
        ^   | (bad-continuation)
clar/data/artificial.py:83:65: W0613: Unused argument 'seed' (unused-argument)
clar/data/artificial.py:84:8: W0613: Unused argument 'meg' (unused-argument)
clar/data/artificial.py:84:18: W0613: Unused argument 'eeg' (unused-argument)
clar/data/artificial.py:122:8: W0613: Unused argument 'n_active' (unused-argument)
clar/data/artificial.py:125:16: W0612: Unused variable 'n_sources' (unused-variable)
clar/data/artificial.py:144:17: W0613: Unused argument 'meg' (unused-argument)
clar/data/artificial.py:144:27: W0613: Unused argument 'eeg' (unused-argument)
clar/data/artificial.py:194:7: R1714: Consider merging these comparisons with "in" to 'n_channels in (M.shape[0], -1)' (consider-using-in)
************* Module clar.data.__init__
clar/data/__init__.py:1:0: R0801: Similar lines in 2 files
==clar.data.artificial:145
==clar.generate_data:68
    if dictionary_type == 'Toeplitz':
        X = get_toeplitz_dictionary(
            n_channels=n_channels, n_sources=n_sources, rho=rho, seed=seed)
    elif dictionary_type == 'Gaussian':
        X = rng.randn(n_channels, n_sources)
    else:
        raise NotImplementedError("No dictionary '{}' in maxsparse"
                                  .format(dictionary_type))
    normalize(X)
    return X


def normalize(X):
    for i in range(X.shape[1]):
        X[:, i] /= norm(X[:, i])
    return X


def get_toeplitz_dictionary(
        n_channels=20, n_sources=30, rho=0.3, seed=0): (duplicate-code)
clar/data/__init__.py:1:0: R0801: Similar lines in 2 files
==clar.data.artificial:122
==clar.generate_data:45
    rng = check_random_state(seed)
    XB = X @ B_star
    n_channels, n_sources = X.shape
    _, n_times = B_star.shape
    # creates the noise
    noise_all_epochs = np.empty((n_epochs, n_channels, n_times))
    for l in range(n_epochs):
        noise = S_star @ rng.randn(n_channels, n_times)
        noise_all_epochs[l, :, :] = noise
    denom = np.sqrt((noise_all_epochs ** 2).sum(axis=(1, 2))).mean()
    multiplicativ_factor = norm(XB, ord='fro') / denom
    multiplicativ_factor /= SNR
    noise_all_epochs *= multiplicativ_factor
 (duplicate-code)
clar/data/__init__.py:1:0: R0801: Similar lines in 2 files
==clar.data.artificial:0
==clar.generate_data:0
import numpy as np

from numpy.linalg import norm
from scipy.linalg import toeplitz
from sklearn.utils import check_random_state


def get_data_me( (duplicate-code)
clar/data/__init__.py:1:0: R0801: Similar lines in 2 files
==clar.data.artificial:185
==clar.generate_data:88
    rng = check_random_state(seed)
    vect = rho ** np.arange(n_sources)
    covar = toeplitz(vect, vect)
    X = rng.multivariate_normal(np.zeros(n_sources), covar, n_channels)
    return X (duplicate-code)

------------------------------------------------------------------
Your code has been rated at 9.16/10 (previous run: 8.91/10, +0.25)
```

File: clar/utils.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
 
 @njit
 def sqrtm(ZZT):
-    """Take the square root of a symetrique definite matrix.
```
Comment (mathurinm) [inline clar/utils.py:11]: <3

**Comment by mathurinm** [comment]
https://github.com/QB3/CLaR/blob/master/clar/tests/test_mrce.py#L3

SNR > snr

https://github.com/QB3/CLaR/blob/master/clar/tests/test_mrce.py#L29 > could we use default aprameters and not specify all of these?

**Comment by QB3** [comment]
Here are the remaining erors when running pylint clar
```
************* Module clar.solvers
solvers.py:200:20: E1101: Module 'numpy.linalg' has no 'pinvh' member; maybe 'pinv'? (no-member)
solvers.py:208:4: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
solvers.py:148:0: R0912: Too many branches (35/12) (too-many-branches)
solvers.py:148:0: R0915: Too many statements (121/50) (too-many-statements)
solvers.py:403:17: E1101: Module 'numpy.linalg' has no 'pinvh' member; maybe 'pinv'? (no-member)
solvers.py:424:29: I1101: Module 'sklearn.linear_model.cd_fast' has no 'enet_coordinate_descent_gram' member, but source is unavailable. Consider adding this module to extension-pkg-whitelist if you want to perform analysis based on run-time introspection of living objects. (c-extension-no-member)
************* Module clar.utils
utils.py:290:37: W0622: Redefining built-in 'ord' (redefined-builtin)
utils.py:296:49: W0622: Redefining built-in 'ord' (redefined-builtin)
```

**Pull request merged**